src/main/src/main/java/org/geoserver/security/impl/LayerGroupContainmentCache.java

```diff
@@ -190,6 +195,14 @@ private void collectContainers(LayerGroupSummary lg, Set<LayerGroupSummary> grou
         }
     }
 
+    @Override
+    public void onApplicationEvent(ApplicationEvent applicationEvent) {
+        if (applicationEvent instanceof ContextRefreshedEvent) {
+            SecureCatalogImpl secureCatalog = GeoServerExtensions.bean(SecureCatalogImpl.class);
+            secureCatalog.getResourceAccessManager().buildLayerGroupCache();
+        }
```

Comment:
we don't need this logic. The LayerGroupContainment cache is a singleton bean now. You can directly invoke the buildLayerGroupCache method no need to go through the SecureCatalogImpl and the AccessManager.